openedx_tagging/core/tagging/models.py

```diff
@@ -409,6 +413,48 @@ def tag_object(
 
         return updated_tags
 
+    def autocomplete_tags(self, search: str, object_id: str = None) -> models.QuerySet:
+        """
+        Provides auto-complete suggestions by matching the `search` string against existing
+        ObjectTags linked to the given taxonomy. A case-insensitive search is used in order
+        to return the highest number of relevant tags.
+
+        If `object_id` is provided, then object tag values already linked to this object
+        are omitted from the returned suggestions. (ObjectTag values must be unique for a
+        given object + taxonomy, and so omitting these suggestions helps users avoid
+        duplication errors.).
+
+        Returns a QuerySet of dictionaries containing distinct `value` (string) and `tag`
+        (numeric ID) values, sorted alphabetically by `value`.
```

Comment:
What do you think about calling it `tag_id` instead of `tag`, if it's a numeric ID?

Reply:
Yes, make sense, updated [here](https://github.com/openedx/openedx-learning/pull/59/commits/7c30263a4b5774ccb9c018ec9421b5f3d6022018#diff-204875f41064da3abfd4648611ad41c6f21abe09277263410e3daafbd343f2a3R457)